adcc/OneParticleOperator.py

```diff
@@ -52,41 +58,210 @@ def __init__(self, spaces, is_symmetric=True):
+
+    def is_zero_block(self, block):
+        """
+        Checks if block is explicitly marked as zero block.
+        Returns False if the block does not exist.
+        """
+        if block not in self.blocks:
+            return False
+        return block not in self.blocks_nonzero
+
+    def block(self, block):
+        """
+        Returns tensor of the given block.
+        Does not create a block in case it is marked as a zero block.
+        Use __getitem__ for that purpose.
+        """
+        if block not in self.blocks_nonzero:
+            raise KeyError("The block function does not support "
+                           "access to zero-blocks. Available non-zero "
+                           f"blocks are: {self.blocks_nonzero}.")
+        return self._tensors[block]
+
+    def __getitem__(self, block):
+        if block not in self.blocks:
+            raise KeyError(f"Invalid block {block} requested. "
+                           f"Available blocks are: {self.blocks}.")
+        if block not in self._tensors:
+            sym = libadcc.make_symmetry_operator(
+                self.mospaces, block, self.is_symmetric, "1"
+            )
+            self._tensors[block] = Tensor(sym)
+        return self._tensors[block]
+
+    def __getattr__(self, attr):
+        from . import block as b
+        return self.__getitem__(b.__getattr__(attr))
+
+    def __setitem__(self, block, tensor):
+        """
+        Assigns a tensor to the specified block
+        """
+        if block not in self.blocks:
+            raise KeyError(f"Invalid block {block} assigned. "
+                           f"Available blocks are: {self.blocks}.")
+        s1, s2 = split_spaces(block)
+        expected_shape = (self.mospaces.n_orbs(s1),
+                          self.mospaces.n_orbs(s2))
+        if expected_shape != tensor.shape:
+            raise ValueError("Invalid shape of incoming tensor. "
+                             f"Expected shape {expected_shape}, but "
+                             f"got shape {tensor.shape} instead.")
+        self._tensors[block] = tensor
+
+    def __setattr__(self, attr, value):
+        try:
+            from . import block as b
+            self.__setitem__(b.__getattr__(attr), value)
+        except AttributeError:
+            super().__setattr__(attr, value)
+
+    def set_block(self, block, tensor):
+        """
+        Assigns tensor to a given block. Deprecated
+        """
+        warnings.warn("The set_block function is deprecated and will be "
+                      "removed in 0.16.0. "
+                      "Use __setitem__ instead.")
+        self.__setitem__(block, tensor)
+
+    def set_zero_block(self, block):
+        """
+        Set a given block as zero block
+        """
+        if block not in self.blocks:
+            raise KeyError(f"Invalid block {block} set as zero block. "
+                           f"Available blocks are: {self.blocks}.")
+        self._tensors.pop(block)
```

Comment:
I wonder if we should have the special syntax
```python
operator.oo = 0
```
for this purpose or is that too much "magic"? Alternative is of course `operator.set_zero_block("o1o1")`.

Reply:
Actually I think that `set_zero_block` is used nowhere, so then we can just leave it as is, I guess?

Reply:
Yes, let's leave it as is. The `dm.oo = 0` syntax would cause confusion, because you'd expect to zero the tensor elements, but not delete the entire block 👍🏻